src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -208,7 +226,7 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
   ): TrezorTypes.CardanoOutput => {
     const changeAddressParams = getAddressParameters(changeOutputFiles, output.address, network)
     const address = encodeAddress(output.address)
-    const tokenBundle = prepareTokenBundle(output.tokenBundle)
+    const tokenBundle = prepareTokenBundle(output.tokenBundle, false)
 
     if (changeAddressParams) {
```

Comment:
The security policies in Ledger and Trezor do not allow outputs given by parameters for SCRIPT_TRANSACTION (because it's not possible to determine, or even define, which outputs are change outputs). In that mode, all outputs have to be given via an opaque address.

Reply:
I am not sure I understand what you mean by that.

Reply:
I'm saying that signing mode affects how the output is sent to HW wallet. For multisig, all output addresses *must* be sent as byte strings. So we should not call prepareChangeOutput in that case (or it should reflect singing mode and return null, and I don't think it does, it should do so in a much more transparent way).

Reply:
Added here https://github.com/vacuumlabs/cardano-hw-cli/pull/108/files#diff-f91c370b4a0d3aba3dde7c9ac1af1e0a2a643f066593823207cc9620e150cbc8R179